forgerock-auth/src/main/java/org/forgerock/android/auth/RemoteDeviceBindingRepository.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2023 ForgeRock. All rights reserved.
+ *
+ *  This software may be modified and distributed under the terms
+ *  of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth
+
+import android.net.Uri
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.coroutines.withContext
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.Response
+import org.forgerock.android.auth.ServerConfig.ACCEPT_API_VERSION
+import org.forgerock.android.auth.devicebind.DeviceRepository
+import org.forgerock.android.auth.devicebind.UserKey
+import org.forgerock.android.auth.exception.ApiException
+import java.io.IOException
+import java.net.HttpURLConnection
+import java.net.URL
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+private val TAG = RemoteDeviceBindingRepository::class.java.simpleName
+
+internal class RemoteDeviceBindingRepository(val serverConfig: ServerConfig = Config.getInstance().serverConfig) :
+    DeviceRepository {
```

Comment:
interface name can be DeviceBindingRepository ?

Reply:
Updated